portal/static/js/main.js

```diff
@@ -4875,106 +909,95 @@ var tnthAjax = {
-        var co = [];
-        var OT = this.getOrgTool();
+        var userId = $("#fillOrgs").attr("userId"), co = [], OT = this.getOrgTool();
         $("#userOrgs input[name='organization']").each(function() {
             if ($(this).is(":checked")) {
                 var po = OT.getElementParentOrg(this);
                 co.push($(this).val());
-                if (hasValue(po)) co.push(po);
-            };
+                if (po) co.push(po);
+            }
         });
-        //exclude currently selected orgs
-        tnthAjax.deleteConsent(userId, {org: "all", exclude: co.join(",")});
+        tnthAjax.deleteConsent(userId, {org: "all", exclude: co.join(",")}); //exclude currently selected orgs
     },
     handleConsent: function(obj) {
-        var self = this;
-        var OT = this.getOrgTool();
-        var userId = $("#fillOrgs").attr("userId");
-        if (!hasValue(userId)) {
-            userId = $("#userOrgs").attr("userId");
-        }
+        var self = this, OT = this.getOrgTool(), userId = $("#fillOrgs").attr("userId") || $("#userOrgs").attr("userId");
         var configVar = $("#profile_CONSENT_WITH_TOP_LEVEL_ORG").val();
-        if (!hasValue(configVar)) {
+        if (!configVar) {
             tnthAjax.getConfigurationByKey("CONSENT_WITH_TOP_LEVEL_ORG", userId, {sync: true}, false, true);
         }
         $(obj).each(function() {
             var parentOrg = OT.getElementParentOrg(this);
             var orgId = $(this).val();
             var cto = String(configVar).toLowerCase() === "true";
-            if ($(this).prop("checked")){
+            if ($(this).prop("checked")) {
                 if ($(this).attr("id") !== "noOrgs") {
                     if (parentOrg) {
                         var agreementUrl = $("#" + parentOrg + "_agreement_url").val();
                         if (agreementUrl && agreementUrl != "") {
-                            var params = CONSENT_ENUM["consented"];
+                            var params = self.consentParams;
                             params.org = cto ? parentOrg : orgId;
                             params.agreementUrl = agreementUrl;
                             setTimeout(function() {
-                                tnthAjax.setConsent($('#fillOrgs').attr('userId'),params, 'all', true, function() {
+                                tnthAjax.setConsent($("#fillOrgs").attr("userId"), params, "all", true, function() {
                                     tnthAjax.removeObsoleteConsent();
                                 });
                             }, 350);
                         } else {
                             if (cto) {
                                 tnthAjax.setDefaultConsent(userId, parentOrg);
-                            };
-                        };
-                    };
-
+                            }
+                        }
+                    }
                 } else {
                     if (cto) {
```

Comment:
Might be more concise to only set the variable you're looping over inside the if/else, and move the loop code outside